### PR TITLE
fix: Conditions for security scan results reporting

### PR DESCRIPTION
**Pull request by mmcgehee-ccc**
Also included:

feat: Optionally specify govulncheck version to use
refactor: Simplify use of Go version from go mod file
fix: Use latest stable Go version to install govulncheck

### CODE REVIEW

**Comment by mmcgehee-ccc** [comment]
@athomas-ccc @jwatson-ccc 
I tested the updated security scan workflow for a variety of conditions using the following projects:
- ga-image-app
- loan-consolidation-webapp
- nes-ocr-app
- cpp-aggregator-service

I can walk you through the various results if you're interested.

File: .github/workflows/golang-security-scan.yml [hunk 1]
```diff
@@ -99,21 +104,26 @@ jobs:
...
           echo "REPORT_PATH=govulncheck-report-${{ matrix.checkout-ref == '' && '1' || '2' }}.txt" >> $GITHUB_ENV
 
+      - name: Set up Go for govulncheck
+        uses: actions/setup-go@v5
+        with:
+          go-version: "stable"
+          cache: false
+
+      - name: Install govulncheck
+        run: go install golang.org/x/vuln/cmd/govulncheck@latest
```
Comment (athomas-ccc) [inline .github/workflows/golang-security-scan.yml:114]: I'm puzzled as to why there is a `govulncheck-version` input added above, but on this line, it isn't using that new input. Is this supposed to be something along these lines?
```suggestion
        run: go install golang.org/x/vuln/cmd/govulncheck@{{$govulncheck-version}}
```

File: .github/workflows/golang-security-scan.yml [hunk 2]
```diff
@@ -99,21 +104,26 @@ jobs:
...
-      - uses: actions/setup-go@v5
+      - name: Set up Go for app
+        uses: actions/setup-go@v5
         with:
-          go-version: ${{ env.GO_VERSION }}
-
-      - name: Install govulncheck
-        run: go install golang.org/x/vuln/cmd/govulncheck@latest
-        shell: bash
+          go-version-file: ${{ inputs.working-directory }}/go.mod
```
Comment (athomas-ccc) [inline .github/workflows/golang-security-scan.yml:125]: Nice! I like how you simplified the way we are getting the go version.

**Review by athomas-ccc — CHANGES_REQUESTED** [review]
One comment I'd like to see addressed before I approve.

File: .github/workflows/golang-security-scan.yml [hunk 3]
```diff
@@ -266,6 +278,7 @@ jobs:
     name: Create issues for failed jobs
     runs-on: ubuntu-latest
     strategy:
+      fail-fast: false
```
Comment (athomas-ccc) [inline .github/workflows/golang-security-scan.yml:281]: Could you elaborate on why this is a helpful change?

**Review by athomas-ccc — APPROVED** [review]
Approved, with one curiosity question!

File: .github/workflows/golang-security-scan.yml (hunk 3 above)
Comment (mmcgehee-ccc) [reply .github/workflows/golang-security-scan.yml:281]: We're using a matrix strategy here, so each matrix label runs separately in parallel (see attached image). The default behavior is `fail-fast: true`, which means if one of the matrix runs fails then the others will be canceled. However, if one fails for some reason, we would prefer that the others still continue to run.

image

**Pull request merged**